Dockerfile

```diff
@@ -1,40 +1,78 @@
-FROM --platform=linux/amd64 ruby:3.0.6
-RUN rm /bin/sh && ln -s /bin/bash /bin/sh
-WORKDIR /usr/src/app
-EXPOSE 3100
+ARG RUBY_VERSION=3.0.6
+FROM registry.docker.com/library/ruby:$RUBY_VERSION-slim as base
 
-RUN apt-get update -y && apt-get install -y --no-install-recommends \
-  default-jre \
-  default-mysql-client \
-  imagemagick \
-  libprotobuf-dev \
-  protobuf-compiler 
+# Rails app lives here
+WORKDIR /rails
```

Comment:
I would recommend that we install all server packages together at one time rather than separate times. Probably after creating the working directories we can do this: 

# Install system dependencies
RUN apt-get update -qq && apt-get install --no-install-recommends -y \
    curl \
    libjemalloc2 \
    libcurl4-openssl-dev \
    default-libmysqlclient-dev \
    # Clean up in the same layer to reduce image size
    && apt-get clean \
    && rm -rf /var/lib/apt/lists/*
    
    
We can also add apt-get clean and removed the apt cache (/var/lib/apt/lists/*) in the same RUN command where packages are installed to further reduce image size.

Reply:
Are you suggesting to install this packages together in here https://github.com/GSA/search-gov/pull/1528/files#diff-dd2c0eb6ea5cfc6c4bd4eac30934e2d5746747af48fef6da689e85b752f39557R29 ?

If so, the idea is that the second packages are only installed in the `build` stage and not in the `base`. That's why they have to be installed separately.

Reply:
@stevenbarragan  - No, what I am saying is that after the build stage, we add the following two lines to the build dependencies: 

```
  && apt-get clean \
    && rm -rf /var/lib/apt/lists/*
```

This will clean and remove the apt cache within the same RUN command

Reply:
I wasn't able to add the second part `&& rm -rf /var/lib/apt/lists/*` because this line https://github.com/GSA/search-gov/pull/1528/files#diff-dd2c0eb6ea5cfc6c4bd4eac30934e2d5746747af48fef6da689e85b752f39557R32 breaks but `apt-get clean` worked just fine